common/src/main/java/com/wynntils/models/spells/SpellModel.java

```diff
@@ -36,6 +44,8 @@ public class SpellModel extends Model {
     private final SpellSegment spellSegment = new SpellSegment();
 
     private SpellDirection[] lastSpell = SpellDirection.NO_SPELL;
+    private SpellDirection[] spellPrediction = SpellDirection.NO_SPELL;
```

Comment:
What is the purpose of this spell prediction? How does it "predict"?

Reply:
It predicts using only the user (and quick cast) inputs for more up-to-date information. instead of using the server packets.